packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -585,21 +953,78 @@ class RoundSliderThumbShape extends SliderComponentShape {
       end: sliderTheme.thumbColor,
     );
     canvas.drawCircle(
-      thumbCenter,
+      center,
       radiusTween.evaluate(enableAnimation),
       Paint()..color = colorTween.evaluate(enableAnimation),
     );
   }
 }
 
+/// This is the default shape to a [Slider]'s thumb overlay if no
+/// other shape is specified.
+///
+/// The shape of the overlay is a circle with the same center as the thumb, but
```

Comment:
Done, one of the TODOs above to make it to that the overlay actually paints on top of the thumb, since that is probably the more logical behavior when painting custom overlays. The default material overlay painter would also be updated to paint an annulus instead of a circle.